src/views/Graph.vue

```diff
@@ -641,11 +1057,255 @@ export default {
+      let removedEdges = []
+
+      // For all cycles...
+      this.cycleArrayStore.forEach((cycle) => {
+        // ...loop through collapsed families...
+        this.collapseFamily.forEach((family) => {
+          // ...get the node from the index...
+          const indexSearch = Object.values(this.cylcTree.$index).find((node) => {
+            return node.name === family && node.tokens.cycle === cycle
+          })
+          // ...now we have a node - we have to understand all its relations in the graph...
+          if (indexSearch) {
+            // ---------------REMOVE NODES BASED ON FAMILY------------
+            // ṃust do this before removing nodes and edges based on cycle as
+            // cycle collapsing takes priority of families
+            // ...this node is collapsed so need to remove any of its children (nodes and edges) from the graph if it has any...
+            this.allChildrenLookUp[indexSearch.id].forEach((config) => {
+              if (config.name !== indexSearch.name && config.name !== '01') {
+                // REMOVE NODES
+                nodes = this.removeNode(config.name, cycle, nodes)
+                // REMOVE EDGES
+                // if there is an edge with a source or target it needs removing
+                const edgeCheckSource = this.checkForEdgeBySource(config.name, cycle, edges)
+                const edgeCheckTarget = this.checkForEdgeByTarget(config.name, cycle, edges)
+
+                if (edgeCheckSource.length) {
+                  [edges, removedEdges] = this.removeEdgeBySource(edgeCheckSource, edges, removedEdges, config, cycle)
+                }
+                if (edgeCheckTarget.length) {
+                  [edges, removedEdges] = this.removeEdgeByTarget(edgeCheckTarget, edges, removedEdges)
+                }
+              }
+            })
+            // ...now we have removed any parts of child nodes that shouldnt be there we can add nodes and edges that should be...
+            // ---------------ADD NODES BASED ON FAMILY------------
+            if (!this.collapseCycle.includes(cycle)) { // cycle collapsing takes priority over family collapsing
+              nodes.push(indexSearch)
+            }
+            const isNodeCollapsedByFamily = (nodeFirstFamily) => {
+              // the nodes first parent is collapsed
+              const firstParent = this.collapseFamily.includes(nodeFirstFamily)
+              // a family member up the tree is collapsed
+              const ancestor = this.allParentLookUp[nodeFirstFamily].some(element => {
+                return this.collapseFamily.includes(element)
+              })
+              if (firstParent && !ancestor) {
+                // the node is collapsed by its first parent
+                return nodeFirstFamily
+              } else if (ancestor) {
+                return this.allParentLookUp[nodeFirstFamily][0]
+                // this is almost certainly an over simplification -> better logic needed here
```

Comment:
TODO

I'm not 100% on what this function does, but I suspect it's missing a bit.

E.g, for this inheritance:

* A
  * A1
    * A11
      * A111

And this state:

| Fam | iscollapsed |
| --- |--- |
| A | :x: |
| A1 | :heavy_check_mark: |
| A11 | :heavy_check_mark: |
| A111 | :x: |

What family should be returned?